smarty/sampler.py

```diff
@@ -317,61 +367,80 @@ def AtomDecorator(self, atom1type, decorator):
             # No alpha atom so the decorator goes before the ']'
             proposed_atomtype = atom1type[0][:-1] + decorator[0] + ']'
             proposed_typename = atom1type[1] + ' '  + decorator[1]
-        return proposed_atomtype, proposed_typename
+
+        return (proposed_atomtype, proposed_typename)
 
     def PickAnAtom(self, atomList):
         """
-        takes a desired current set of atomtypes (current, base, populated base, or some combination of the
-        above or something else) and pick one at random. (Here, we will typically but perhaps not always want
-        to use currently populated atomtypes, i.e. the atomtypes list, not basetypes (generic types) or
-        used_basetypes (base types which type anything)). A little silly, but still helpful.
-        Returns a tuple corresponding to the selected (smarts, name).
+        Parameters
+        ----------
+        atomList : any list of tuples in the form (smarts, typename)
+                   this could include decorator or bond lists
+
+        Returns
+        -------
+        one random (smarts, typename) pair from given list
+
+        This allows for continuity in the code, this method could be changed,
+        and all random choices would still be made in the same way.
+        It also allowed for testing which atomtypes to choose from while sampling.
         """
-        atomtype_index = random.randint(0, len(atomList)-1)
-        return atomList[atomtype_index]
+        return random.choice(atomList)
 
     def HasAlpha(self, atom1type):
         """
-        Takes a specified atomtype tuple (smarts, name) and determines whether or not it already has an alpha
-        substituent, returning True or False.
+        Parameter
+        ---------
+        atom1type : an atomtype tuple (smarts, typename)
+
+        Returns
+        -------
+        True if atomtype has at least 1 alpha substituent otherwise False
         """
+        # TODO: check does this work if you're using replacements
+        # CCB: I don't think it will work!
```

Comment:
@davidlmobley 
I was going to check this before writing an issue. Since we aren't currently using any of the replacement strings in our initial types I didn't think it needed to be addressed in this pull request.